### PR TITLE
Add note about Jackson configuration in rest client guides

### PR DESCRIPTION
**Pull request by geoand**
Originally mentioned in https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/Mapping.20complex.20structures.20with.20rest.20client/near/266014391

### CODE REVIEW

**Comment by geoand** [comment]
cc @galderz

File: docs/src/main/asciidoc/rest-client-reactive.adoc [hunk 1]
```diff
@@ -94,6 +94,11 @@ public class Extension {
 
 The model above is only a subset of the fields provided by the service, but it suffices for the purposes of this guide.
 
+[NOTE]
+====
+The fields of `Extension` where made `public` in order to make Jackson to be able to serialize them using its default settings. For more information about configuring Jackson, see xref:rest-json.adoc#jackson[this guide].
```
Comment (gsmet) [inline docs/src/main/asciidoc/rest-client-reactive.adoc:99]:
```suggestion
The fields of `Extension` were made `public` in order to make Jackson able to serialize them using its default settings. For more information about configuring Jackson, see xref:rest-json.adoc#jackson[this guide].
```

File: docs/src/main/asciidoc/rest-client.adoc [hunk 2]
```diff
@@ -97,6 +97,11 @@ public class Extension {
 
 The model above is only a subset of the fields provided by the service, but it suffices for the purposes of this guide.
 
+[NOTE]
+====
+The fields of `Extension` where made `public` in order to make Jackson to be able to serialize them using its default settings. For more information about configuring Jackson, see xref:rest-json.adoc#jackson[this guide].
```
Comment (gsmet) [inline docs/src/main/asciidoc/rest-client.adoc:102]:
```suggestion
The fields of `Extension` were made `public` in order to make Jackson able to serialize them using its default settings. For more information about configuring Jackson, see xref:rest-json.adoc#jackson[this guide].
```

**Review by gsmet — CHANGES_REQUESTED** [review]
Thanks. Spotted a few typos.

**Pull request closed without merging**